src/net/netdev.go

```diff
@@ -0,0 +1,52 @@
+package net
+
+import (
+	"time"
+)
+
+const (
+	// Made up, not a real IP protocol number.  This is used to create a
+	// TLS socket on the device, assuming the device supports mbed TLS.
+	IPPROTO_TLS = 0xFE
+)
+
+// netdev is the current netdev, set by the application with useNetdev()
+var netdev netdever
+
+// (useNetdev is go:linkname'd from tinygo/drivers package)
+func useNetdev(dev netdever) {
+	netdev = dev
+}
+
+// Netdev is TinyGo's network device driver model.  Network drivers implement
+// the netdever interface, providing a common network I/O interface to TinyGo's
+// "net" package.  The interface is modeled after the BSD socket interface.
+// net.Conn implementations (TCPConn, UDPConn, and TLSConn) use the netdev
+// interface for device I/O access.
+//
+// A netdever is passed to the "net" package using net.useNetdev().
+//
+// Just like a net.Conn, multiple goroutines may invoke methods on a netdever
+// simultaneously.
+//
+// NOTE: The netdever interface is mirrored in drivers/netdev.go.
+// NOTE: If making changes to this interface, mirror the changes in
+// NOTE: drivers/netdev.go, and visa-versa.
+
+type netdever interface {
+
+	// GetHostByName returns the IP address of either a hostname or IPv4
+	// address in standard dot notation
+	GetHostByName(name string) (IP, error)
+
+	// Berkely Sockets-like interface, Go-ified.  See man page for socket(2), etc.
+	Socket(domain int, stype int, protocol int) (int, error)
+	Bind(sockfd int, ip IP, port int) error
```

Comment:
Could we use `uintptr`s for file descriptors just like the go standard library? Since we are returning an error there is no need for out-of-band error flagging so no need for them to be signed.

Reply:
@soypat Thank you for the review.

I'm not seeing where the standard Go library uses uintptr for file descriptors.  Go's syscall package uses int for fds ([see, for example](https://pkg.go.dev/syscall#Open)).  Additionally, the POSIX and BSD socket(2) calls use ints for fds.

Reply:
Fair enough, I thought the use of uintptrs was more widespread